helm/chart/router/templates/configmap.yaml

```diff
@@ -1,6 +1,10 @@
 {{- if .Values.router.configuration }}
```

Comment:
I'm unsure if this set of changes is correct. If I run 
```
helm template --dry-run --debug helm/chart/router
```
then I get a set of templates which include a ConfigMap which looks like this:
```
data:
  configuration.yaml: |
    health_check:
      listen: 0.0.0.0:8088
    supergraph:
      listen: 0.0.0.0:4000
    telemetry:
      metrics:
        common:
          resources:
            service.name: release-name-router
```
That surprised me a little, since I wouldn't expect to get any telemetry configuration if I haven't specified any.

I modified the ConfigMap template like this:
```
{{/* NOTE: metrics configuration moved under telemetry.exporters in Router 1.35.0 */}}
{{- $configuration := dict }}
{{- if (.Values.router.configuration.telemetry).exporters }}
{{- $configuration = dict "telemetry" (dict "exporters" (dict "metrics" (dict "common" (dict "resource" (dict "service.name" $routerFullName))))) -}}
{{- else }}
{{- $configuration := dict "telemetry" (dict "metrics" (dict "common" (dict "resources" (dict "service.name" $routerFullName)))) -}}
{{- end }}
```
and that gave me a ConfigMap that looks like this:
```
data:
  configuration.yaml: |
    health_check:
      listen: 0.0.0.0:8088
    supergraph:
      listen: 0.0.0.0:4000
```
which is what I would expect.

What I don't fully understand is why your change is always generating a dict which contains a "service.name". 

My theory is that this logic was always broken but "worked" because we always specified some telemetry information in the values.yaml file (which you have now removed).

Let's talk this through when you've had a chance to read through this.

Reply:
This all makes sense, I've included your suggested change.

I tested this change against the in house router's chart in three ways:
* As it currently is, using the deprecated shape. The results are the same and the router can start up with it
* Without any telemetry at all — the resulting config no longer contains `telemetry`.
* After updating telemetry configuration to the new structure — the resulting config is correct and the router no longer spits out deprecation warnings

> My theory is that this logic was always broken but "worked" because we always specified some telemetry information in the values.yaml file (which you have now removed).

I think this is correct